### PR TITLE
Feat/Add SQLMesh UI Support

### PR DESCRIPTION
**Pull request by pedroheyerdahl**
This PR adds support for SQLMesh's web UI interface

## Key Changes
- Added UI service to docker-compose.yml using common configuration
- Updated entrypoint.sh to handle UI command
- Added web dependencies to requirements.txt
- Added UI_PORT configuration to .env.example

## Important Caveat
The UI service must be run using the dedicated profile flag:
```bash
docker compose --profile ui up ui
```
Do NOT use `docker compose run pipeline ui` as it won't work with the `--gateway` flag. The UI command has different CLI options than other SQLMesh commands.

## Testing
1. Copy UI_PORT to your .env file
2. Run `docker compose --profile ui up ui`
3. Access UI at http://localhost:8000 (or your configured port)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
-    command: etl
+      - GATEWAY=${GATEWAY:-local}
+    profiles:
+      - default
+
+  ui:
+    <<: *common-config
+    container_name: osaa-mvp-ui
+    entrypoint: ["sqlmesh"]
+    command: ["ui", "--port", "${UI_PORT:-8000}"]
```
Comment (risafangs) [inline docker-compose.yml:31]: in the `.env.example` it uses `8080`, should this be the same for consistency/simplicity?

Comment (pedroheyerdahl) [reply docker-compose.yml]: thanks Lisa, I've change it to be consistent with env.example

File: requirements.txt [hunk 2]
```diff
@@ -5,6 +5,10 @@ duckdb==1.1.3
 fsspec>=2024.2.0
 ibis-framework[duckdb]==9.5.0
 mypy==1.8.0
+numpy<2.0.0
```
Comment (risafangs) [inline requirements.txt:8]: it looks like there’s a specific version of numpy on line 11/15 below, is this duplicative?

Comment (pedroheyerdahl) [reply requirements.txt:8]: yes, removing this one. thx

**Pull request merged**